### PR TITLE
DEVPROD-3889 remove customizable logging

### PR DESCRIPTION
**Pull request by ablack12**
DEVPROD-3889
### Description
This "quick win" removes customizable logging. Not sure if more infrastructure here should be removed. 

### Testing
Ran a task in staging, verifying that no projects are using it, and updated unit tests.

### Documentation
Removed the docs.

### CODE REVIEW

File: agent/internal/client/base_client.go [hunk 1]
```diff
@@ -372,109 +372,57 @@ func (c *baseCommunicator) GetLoggerProducer(ctx context.Context, tsk *task.Task
...
 }
 
-func (c *baseCommunicator) makeSender(ctx context.Context, tsk *task.Task, opts []LogOpts, config *LoggerConfig, logType taskoutput.TaskLogType) (send.Sender, []send.Sender, error) {
+func (c *baseCommunicator) makeSender(ctx context.Context, tsk *task.Task, config *LoggerConfig, logType taskoutput.TaskLogType) (send.Sender, error) {
 	levelInfo := send.LevelInfo{Default: level.Info, Threshold: level.Debug}
 	var senders []send.Sender
 	if config.SendToGlobalSender {
 		senders = append(senders, grip.GetSender())
 	}
-	underlyingBufferedSenders := []send.Sender{}
```
Comment (malikchaya2) [inline agent/internal/client/base_client.go:407]: should we remove [this part](https://github.com/evergreen-ci/evergreen/blob/main/agent/internal/client/logger_producer.go#L69) as well?

**Review by malikchaya2 — CHANGES_REQUESTED** [review]

File: agent/internal/client/base_client.go (hunk 1 above)
Comment (ablack12) [reply agent/internal/client/base_client.go:407]: good catch!

**Review by malikchaya2 — CHANGES_REQUESTED** [review]
It looks like there are a lot of failing tasks so I’ll hold off re reviewing for now.

**Comment by ablack12** [comment]
Just a missed merge conflict, if you can look again (generally I'd recommend this and you can LGTM conditionally to reduce back-and-forth time on PRs but I'll also try and catch these things faster!)

**Review by malikchaya2 — APPROVED** [review]

**Pull request merged**